htlcswitch/link.go

```diff
@@ -1101,7 +1110,18 @@ func (l *channelLink) htlcManager() {
 			newCommitFee := chainfee.SatPerKWeight(
 				math.Min(float64(netFee), float64(maxFee)),
 			)
-			if !shouldAdjustCommitFee(newCommitFee, commitFee) {
+
+			// If the new commitment fee is below the minimum
+			// relay fee then we bump it to be equal to the minimum
+			// relay fee.
+			minRelayFee := l.cfg.FeeEstimator.RelayFeePerKW()
+			if newCommitFee < minRelayFee {
+				newCommitFee = minRelayFee
+			}
+
+			if !shouldAdjustCommitFee(
+				newCommitFee, commitFee, minRelayFee,
+			) {
```

Comment:
My only worry here is the extreme case where the minRelayFee is larger than the maximum fee we are able to pay. What do we want to do in that case?

Reply:
Ok i have updated it to use max fee in this case